scripts/metadata_comparison/metadata_comparison/extractor.py

```diff
@@ -0,0 +1,219 @@
+    return papi_operation_to_api_mapping
+
+
+def read_genomics_papiv1_operations_metadata(jobId, api, genomics_client):
+    """Reads the operations metadata for a pipelines API v1 job."""
+    return genomics_client.operations().get(name=jobId).execute()
+
+
+def pretty_print_json(raw_json):
+    return json.dumps(json.loads(raw_json), indent=4, separators=(',', ': '))
+
+
+def read_operations_metadata(jobId, api, genomics_client, verbose):
+    """Reads the operations metadata for a pipelines API v2alpha1 job ID. Returns a python dict"""
+    # TODO: Reroute appropriately for different API versions
+    if verbose:
+        print(f'Reading operation metadata for {jobId}...', end='')
+    result = genomics_client.projects().operations().get(name=jobId).execute()
+    if verbose:
+        print(' done!')
+    return result
+
+
+def upload_blob(bucket_name, source_file_contents, destination_blob_name, gcs_storage_client, verbose):
+    """Uploads a file to the cloud"""
+    # bucket_name = "your-bucket-name"
+    # source_file_contents = "... some file contents..."
+    # destination_blob_name = "storage/object/name"
+
+    bucket = gcs_storage_client.bucket(bucket_name)
+
+    if verbose:
+        print(f'Uploading file content to gs://{bucket_name}/{destination_blob_name}...', end='')
+    blob = bucket.blob(destination_blob_name)
+    blob.upload_from_string(source_file_contents)
+    if verbose:
+        print(" done!")
+
+
+def upload_workflow_metadata_json(bucket_name, raw_workflow_metadata, workflow_gcs_base_path, gcs_storage_client, verbose):
+    workflow_gcs_metadata_upload_path = f'{workflow_gcs_base_path}/metadata.json'
+    upload_blob(bucket_name, raw_workflow_metadata, workflow_gcs_metadata_upload_path, gcs_storage_client, verbose)
+
+
+def upload_operations_metadata_json(bucket_name, operation_id, operations_metadata, workflow_gcs_base_path, gcs_storage_client, verbose):
+    """Uploads metadata to cloud storage, as json"""
+    operation_upload_path = f'{workflow_gcs_base_path}/operations/{get_operation_id_number(operation_id)}.json'
+    formatted_metadata = json.dumps(operations_metadata, indent=2)
+    upload_blob(bucket_name, formatted_metadata, operation_upload_path, gcs_storage_client, verbose)
+
+
+def process_workflow(cromwell_url, gcs_bucket, gcs_path, gcs_storage_client, genomics_client, workflow, verbose):
+    raw_metadata, json_metadata = fetch_raw_workflow_metadata(cromwell_url, workflow, verbose)
+    workflow_gcs_base_path = f'{gcs_path}/{workflow}/extractor'
+
+    operation_ids = find_operation_ids_in_metadata(json_metadata)
+    for (id, api) in operation_ids.items():
+        operation_metadata = read_operations_metadata(id, api, genomics_client, verbose)
+        upload_operations_metadata_json(gcs_bucket, id, operation_metadata, workflow_gcs_base_path, gcs_storage_client, verbose)
+    upload_workflow_metadata_json(gcs_bucket, raw_metadata, workflow_gcs_base_path, gcs_storage_client, verbose)
+
+
+
```

Comment:
doesn't IntelliJ's PEP 8 enforcer have something to say about this quantity of blank lines here?

Reply:
This file is certainly not PEP8 compliant. The line widths would have to be 80 or less, for example.

Reply:
That said, I'll shrink the whitespace